server/src/main/java/io/crate/execution/ddl/TransportSchemaUpdateAction.java

```diff
@@ -221,14 +222,14 @@ static void mergeIntoSource(Map<String, Object> source, Map<String, Object> mapp
             String key = updateEntry.getKey();
             Object updateValue = updateEntry.getValue();
             if (source.containsKey(key)) {
-                Object sourceValue = source.get(key);
+                Object sourceValue = Objects.requireNonNull(source.get(key), "schema metadata should not contain null-valued entries");
```

Comment:
Wouldn't this still result in an NPE if a source with existing NULL values (existing mapping inside a running older cluster) is processed?
I think we must handle NULL values here for this BWC reason.

Reply:
Thank you for catching it. It would handle BWC by simply reverting the line you pointed, but I thought it would be more clear to handle it more explicitly.